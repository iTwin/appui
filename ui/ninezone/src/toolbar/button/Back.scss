/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_variables";

.nz-toolbar-button-back {
  border-radius: $mls-button-width / 2;
}
