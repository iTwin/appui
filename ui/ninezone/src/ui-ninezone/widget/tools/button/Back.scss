/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_variables";

.nz-toolbar-button-back {
  border-radius: $mls-button-width / 2;
}
