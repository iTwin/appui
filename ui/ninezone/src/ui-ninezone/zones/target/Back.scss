/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "../../_prefix";

.nz-zones-target-back {
  > .nz-arrow {
    @include nz-backface-visibility(hidden);
  }

  @mixin nz-arrow($zone, $rotation) {
    &.nz-zone-#{$zone} {
      > .nz-arrow {
        transform: rotateZ($rotation * 1deg) translate3d(0, 0, 1px);
      }
    }
  }

  @include nz-arrow(6, 45 * 0);
  @include nz-arrow(9, 45 * 1);
  @include nz-arrow(8, 45 * 2);
  @include nz-arrow(7, 45 * 3);
  @include nz-arrow(4, 45 * 4);
  @include nz-arrow(1, 45 * 5);
  @include nz-arrow(2, 45 * 6);
  @include nz-arrow(3, 45 * 7);
}
