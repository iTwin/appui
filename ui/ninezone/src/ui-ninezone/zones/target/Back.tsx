/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Zone */

import * as classnames from "classnames";
import * as React from "react";
import { WidgetZoneIndex } from "../state/NineZone";
import { Arrow } from "./Arrow";
import { MergeTargetProps } from "./Merge";
import { ZoneTarget } from "./Target";

/** Properties of [[BackTarget]] component.
 * @beta
 */
export interface BackTargetProps extends MergeTargetProps {
  /** Describes back target arrow rotation. */
  zoneIndex: WidgetZoneIndex;
}

/** Zone target used to merge widget back to initial zone.
 * @beta
 */
export class BackTarget extends React.PureComponent<BackTargetProps> {
  public render() {
    const { className, ...props } = this.props;
    const targetClassName = classnames(
      "nz-zones-target-back",
      this.props.className);

    return (
      <ZoneTarget
        className={targetClassName}
        {...props}
      >
        <Arrow zoneIndex={this.props.zoneIndex} />
      </ZoneTarget>
    );
  }
}
