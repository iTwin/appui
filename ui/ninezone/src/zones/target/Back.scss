/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
@import "../../_prefix";

.nz-zones-target-back {
  > .nz-arrow {
    transform: rotateZ(45deg);
    @include nz-backface-visibility(hidden);
  }
}
