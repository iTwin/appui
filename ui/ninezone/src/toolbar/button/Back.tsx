/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
/** @module Toolbar */

import * as classnames from "classnames";
import * as React from "react";

import Icon from "./Icon";
import { ToolbarButtonProps } from "./Button";
import "./Back.scss";

export default class BackButton extends React.Component<ToolbarButtonProps> {
  public render() {
    const { className, ...props } = this.props;
    const buttonClassName = classnames(
      "nz-toolbar-button-back",
      className);

    return (
      <Icon
        className={buttonClassName}
        icon={this.props.children}
        {...props}
      />
    );
  }
}
