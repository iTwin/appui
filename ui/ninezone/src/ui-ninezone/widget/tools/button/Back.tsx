/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Toolbar */

import * as classnames from "classnames";
import * as React from "react";
import { OmitChildrenProp, NoChildrenProps } from "@bentley/ui-core";
import { ToolbarIcon, ToolbarIconProps } from "./Icon";
import "./Back.scss";

/** Properties of [[BackButton]] component.
 * @alpha
 */
export interface BackButtonProps extends OmitChildrenProp<ToolbarIconProps>, NoChildrenProps {
}

/** Back button which displays icon. Used in [[Toolbar]] component.
 * @note See basic button: [[ToolbarButton]]
 * @alpha
 */
export class BackButton extends React.PureComponent<BackButtonProps> {
  public render() {
    const { className, ...props } = this.props;
    const buttonClassName = classnames(
      "nz-toolbar-button-back",
      className);

    return (
      <ToolbarIcon
        className={buttonClassName}
        {...props}
      />
    );
  }
}
