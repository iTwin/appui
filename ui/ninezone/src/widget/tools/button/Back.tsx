/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Toolbar */

import * as classnames from "classnames";
import * as React from "react";
import { OmitChildrenProp, NoChildrenProps } from "../../../utilities/Props";
import Icon, { ToolbarIconProps } from "./Icon";
import "./Back.scss";

/** Properties of [[BackButton]] component. */
export interface BackButtonProps extends OmitChildrenProp<ToolbarIconProps>, NoChildrenProps {
}

/**
 * Back button which displays icon. Used in [[Toolbar]] component.
 * @note See basic button: [[ToolbarButton]]
 */
export default class BackButton extends React.Component<BackButtonProps> {
  public render() {
    const { className, ...props } = this.props;
    const buttonClassName = classnames(
      "nz-toolbar-button-back",
      className);

    return (
      <Icon
        className={buttonClassName}
        {...props}
      />
    );
  }
}
