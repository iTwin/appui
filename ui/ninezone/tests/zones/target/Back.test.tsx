/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";
import Back from "../../../src/zones/target/Back";

describe("<Back />", () => {
  it("should render", () => {
    mount(<Back />);
  });

  it("renders correctly", () => {
    shallow(<Back />).should.matchSnapshot();
  });
});
