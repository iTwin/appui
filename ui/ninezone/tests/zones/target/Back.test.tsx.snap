// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Back /> renders correctly 1`] = `
<Target
  className="nz-zones-target-back"
>
  <Component
    className="nz-arrow"
  />
</Target>
`;
